### PR TITLE
Feature api

### PR DESCRIPTION
**Pull request by dev1980**
>Fetch API to generate weather 
>Generated data is based on input city.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+const api = {
+  key: '80068fbfc8f92310ac9ea24006b4d8f7',
+  base: 'http://api.openweathermap.org/data/2.5/',
+};
+
+const inputValue = document.querySelector('.inputValue');
+inputValue.addEventListener('keypress', setQuery);
+
+function setQuery(e) {
```
Comment (mcrd25) [inline src/index.js]: Ought to use ES6 arrow functions instead of the `function` keyword to create functions

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  if (e.keyCode === 13) {
+    getResults(inputValue.value);
+  }
+}
+
+function getResults(query) {
```
Comment (mcrd25) [inline src/index.js]: Ought to use ES6 arrow functions instead of the `function` keyword to create functions

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  fetch(`${api.base}weather?q=${query}&units=metric&APPID=${api.key}`)
+    .then(weather => weather.json()).then(displayResults);
+}
+
+function clearForm() {
```
Comment (mcrd25) [inline src/index.js]: Ought to use ES6 arrow functions instead of the `function` keyword to create functions

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  setTimeout(() => {
+    inputValue.value = '';
+  }, 3000);
+}
+
+function displayResults(weather) {
```
Comment (mcrd25) [inline src/index.js]: Ought to use ES6 arrow functions instead of the `function` keyword to create functions

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  const weatherNew = document.querySelector('.current .weather');
+  weatherNew.innerText = weather.weather[0].main;
+
+  const hilow = document.querySelector('.hi-low');
+  hilow.innerText = `${Math.round(weather.main.temp_min)}°c / ${Math.round(weather.main.temp_max)}°c`;
+
+  clearForm();
+}
+
+function newDate(d) {
```
Comment (mcrd25) [inline src/index.js]: Ought to use ES6 arrow functions instead of the `function` keyword to create functions

**Review by mcrd25 — CHANGES_REQUESTED** [review]
# Changes Required 🔍 👀 

Hi [Dev](@dev1980) 👋 

Good job with this


#### Changes Required
- you are missing the toggle feature (Between degrees Celsius and Fahrenheit) See this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/search_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app)
- You are also missing the feature that changes appearance based on received data. See this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/appearance_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app)
- As seen from this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/es6_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622983/) You are required to use ES6  syntax in this project. I.E you ought to use arrow functions, `import` `export` etc (See comments for places that need to be updated)
 

#### Additional Suggestions
- [OPTIONAL] Consider adding your screenshot of the app to README (line 14) from now.
- Don't forget to add a live demo link when you've deployed. [OPTIONAL] You may add a temporary demo link using [raw.githack.com](https://rawcdn.githack.com/)

***Please makes changes and submit another review request.
If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

**Comment by mcrd25** [comment]
Hi @dev1980 

Please do not make change just yet! I accidentally submitted the review prematurely! Before finishing

**Comment by mcrd25** [comment]
Hi @dev1980 

Now you can make the changes! 😄

File: src/index.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  const url = `https://api.openweathermap.org/data/2.5/weather?q=${city}&APPID=${KEY}`;
+  fetch(url).then(response => {
+    response.json().then(json => {
+      data = json;
+      getResponse(data);
+    });
+  });
+};
+
+function kToC(kTemp) {
```
Comment (mcrd25) [inline src/index.js]: Oh, man! You did such a great job with the changes! But you forgot to implement this using ES6 syntax. 😢 
Please convert to arrow function syntax

**Review by mcrd25 — CHANGES_REQUESTED** [review]
# changes required 🔍 👀 

Hi Dev!

You did a great job with changes! Please address the two issues left and submit another review. 😄 

#### Required Changes
- You are still missing the feature that changes appearance based on received data. See this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/appearance_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app)
Some ideas: Have the giphy image change depending on weather (eg. if it's clouds a different giphy image, clear a diff giphy image or maybe you can change the color background based on temperature. 
- There is an implementation of a function using `function` keyword instead of ES6 arrow functions

--
Maya

File: src/index.js (hunk 4 above)
Comment (dev1980) [reply src/index.js]: change to ES6 function

**Review by MauricioRobayo — APPROVED** [review]
Congratulations! You successfully addressed all requirements from previous reviews. The project is approved. Move on!

Status: __Approved__.

__OPTIONAL__: You could try to do different requests for Celsius and Fahrenheit to strengthen your API mastery:

* The application makes different requests for the Celsius data and the Fahrenheit data (you could get the data for either unit and use math to convert it to the other unit. That beats the point of the assignment) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/requests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/))

Take a look at the [`units` parameter of the API](https://openweathermap.org/current#data) to do that.

**Pull request merged**